corehq/motech/views.py

```diff
@@ -79,6 +81,23 @@ class MotechLogDetailView(BaseProjectSettingsView, DetailView):
     def get_queryset(self):
         return RequestLog.objects.filter(domain=self.domain)
 
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        result = lookup_doc_id(context['log'].payload_id)
+        doc_link = None
+        if result:
+            doc_info = get_doc_info(result.doc)
+            if doc_info:
+                user = self.request.couch_user
+                has_permission = doc_info.user_has_permissions(context['log'].domain, user, result.doc)
+                if has_permission:
```

Comment:
nit: the extra var here isn't necessary and I don't think it adds value
```suggestion
                if doc_info.user_has_permissions(context['log'].domain, user, result.doc)
```

Reply:
I was going to do that originally, but thought it might make the code less readable by putting the function call there. Will definitely keep in mind for future code changes!